### DISCUSSION TITLE
Crack does not propagate with XFEM 3D

### DISCUSSION THREAD

**Discussion by leeczway** [idaholab/moose#27946] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi developers,
I am using the XFEM module to do pressurized cracking, and it works as expected in 2D case with the userobjects of LineSegmentCutUserObject and XFEMRankTwoTensorMarkerUserObject. But the similar combination of CircleCutUserObject and XFEMRankTwoTensorMarkerUserObject does not produce crack propagation even if the Maxprinciple stress was far exceeded.
Is there a way to do 3D crack propagation simulation without using CrackMeshCut3DUserObject, since it seems cracking direction and speed has to be prescribled with options like growth_direction and growth_rate?
Thanks for your help!
Wei

**Comment by GiudGiud** [comment]
@hugary1995

**Reply by hugary1995** [reply]
I'll have to defer this question to @jiangwen84 @bwspenc @lynnmunday

**Reply by lynnmunday** [reply]
I haven't done much with the 3d xfem code but I think this test does what you want: moose/modules/xfem/test/tests/solid_mechanics_basic/edge_crack_3d_mhs.i
This uses CrackMeshCut3DUserObject to grow the crack.  The growth rate function is just used to grow the crack over that time step to see if it reaches a size where it will no longer grow.  The crack direction is still determined by a fracture integral.  The nucleation objects won't nucleate a new crack in 3D.  You would have to write code to nucleate a small circular crack in 3D.  I think it would follow the steps used to nucleate and grow cracks in 2D so it should be possible.
